src/model/mosaic/Currency.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NamespaceId } from '../namespace/NamespaceId';
+import { UInt64 } from '../UInt64';
+import { Mosaic } from './Mosaic';
+import { MosaicId } from './MosaicId';
+import { UnresolvedMosaicId } from './UnresolvedMosaicId';
+
+/**
+ * An object that knows how to create Mosaics based on the Mosaic Info and Namespace configuration.
+ */
+export class Currency {
```

Comment:
Shall we call it NetworkCurrency instead of Currency? to differentiate with other tokens might be created by users

Reply:
That's the thing, Currency supports user-created mosaics...